### DISCUSSION TITLE
Spotify embed not rendering when published[Bug report]

### DISCUSSION THREAD

**Discussion by Disdainz** [GetPublii/Publii#1804] [Bug Report] [answered]
### Operating system

Windows 11

### Publii version

Version: 0.46.2 (build 16968)

### Issue type

Application

### Bug description

Writing a new page post, using custom HTML I use Spotify generated iFrame code and in the WYSIWIG editor the embed appears and works.

In a preview and when the site is published, the Spotify widget does not appear. There are no related errors in the console and checked hosting for blocked mixed content or similar. (A blank HTML page with the code works fine, just not through Publii).

I can't see anything in the Publii documentation or online that suggests blocking iFrames or javascript, so assume there's a bug somewhere here.

### Steps to reproduce

- Generate Spotify code
EG: `<iframe style="border-radius:12px" src="https://open.spotify.com/embed/album/73XL3h5TbQzBclgbfZP7ry?utm_source=generator" width="100%" height="352" frameBorder="0" allowfullscreen="" allow="autoplay; clipboard-write; encrypted-media; fullscreen; picture-in-picture" loading="lazy"></iframe>`

- Create page in Publii
- Add code using WYSWIG editor
- Preview or Publish
- ???

**Comment by candidexmedia** [comment, marked as answer]
Try adding the `data-responsive="false"` attribute inside the `<iframe>` tag ([docs ref](https://getpublii.com/dev/how-the-post-content-is-modified-before-the-output/#make-iframes-responsive))

So it would look like this:

```html
<iframe 
  data-responsive="false" 
  style="border-radius:12px" 
  src="https://open.spotify.com/embed/album/2lIZef4lzdvZkiiCzvPKj7?utm_source=generator" 
  width="100%" 
  height="352" 
  frameBorder="0" 
  allowfullscreen="" 
  allow="autoplay; clipboard-write; encrypted-media; fullscreen; picture-in-picture" 
  loading="lazy">
</iframe>
```

Similar issues:
- https://forum.getpublii.com/topic/best-way-to-create-forms-for-publii-sites/#post-10613
- https://forum.getpublii.com/topic/audio-player-embed/#post-8414

**Reply by Disdainz** [reply]
Oh wow thank you! Didn't realise that it was transformed on publishing. 

👍
